File: src/main/java/me/kuku/simbot/entity/IqiYiEntity.java
package me.kuku.simbot.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Getter
@Setter
@ToString
@NoArgsConstructor
@Entity
@Table(name = "i_qi_yi")
public class IqiYiEntity {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	@OneToOne
	@JoinColumn(name = "qq")
	private QqEntity qqEntity;
	@Column(length = 1000)
	private String cookie;
	private String pOne;
	private String pThree;

	public IqiYiEntity(QqEntity qqEntity){
		this.qqEntity = qqEntity;
	}

	public IqiYiEntity(String cookie, String pOne, String pThree){
		this.cookie = cookie;
		this.pOne = pOne;
		this.pThree = pThree;
	}
}

File: src/main/kotlin/me/kuku/simbot/entity/IqiYiEntity.kt
package me.kuku.simbot.entity

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service
import javax.annotation.Resource
import javax.persistence.*

@Entity
@Table(name = "i_qi_yi")
data class IqiYiEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,
    @OneToOne
    @JoinColumn(name = "qq")
    var qqEntity: QqEntity? = null,
    @Column(length = 1000)
    var cookie: String = "",
    var pOne: String = "",
    var pThree: String = ""
){
    companion object{
        fun getInstance(qqEntity: QqEntity): IqiYiEntity{
            return IqiYiEntity(qqEntity = qqEntity)
        }
        fun getInstance(cookie: String, pOne: String, pThree: String): IqiYiEntity{
            return IqiYiEntity(cookie = cookie, pOne = pOne, pThree = pThree)
        }
    }
}

interface IqiYiRepository: JpaRepository<IqiYiEntity, Int>{
    fun findByQqEntity(qqEntity: QqEntity): IqiYiEntity?
}

interface IqiYiService{
    fun findByQqEntity(qqEntity: QqEntity): IqiYiEntity?
    fun save(entity: IqiYiEntity): IqiYiEntity
    fun findAll(): List<IqiYiEntity>
    fun delete(entity: IqiYiEntity)
}

@Service
class IqiYiServiceImpl: IqiYiService{

    @Resource
    private lateinit var iqiYiRepository: IqiYiRepository

    override fun findByQqEntity(qqEntity: QqEntity): IqiYiEntity? {
        return iqiYiRepository.findByQqEntity(qqEntity)
    }

    override fun save(entity: IqiYiEntity): IqiYiEntity {
        return iqiYiRepository.save(entity)
    }

    override fun findAll(): List<IqiYiEntity> {
        return iqiYiRepository.findAll()
    }

    override fun delete(entity: IqiYiEntity) {
        iqiYiRepository.delete(entity)
    }
}